R/methods-Megastudy.R

```diff
@@ -0,0 +1,123 @@
+#this first isnt really specific to any class
+#just putting it here bc the megastudy introduced the need and i dont see an obviously better spot
+
+#' VariableSpec as String
+#' 
+#' This function returns a string representation of a VariableSpec. By
+#' default it assumes the VariableSpec is available in a slot called
+#' `variableSpec`.
+#' 
+#' @param object An object containing a veupathUtils::VariableSpec
+#' @return character
+#' @export
+setGeneric("getVariableSpecAsString", 
+  function(object) standardGeneric("getVariableSpecAsString"),
+  signature = "object"
+)
+
+#' @export
+setMethod('getVariableSpecAsString', signature('ANY'), function(object) {
+  if (!'variableSpec' %in% slotNames(object)) stop("Specified object does not have a `variableSpec` slot.")
+
+  return(getColName(object@variableSpec))
+})
+
+#' @export 
+setMethod('getVariableSpecAsString', signature('StudySpecificVocabulariesByVariable'), function(object) {
+  return(getVariableSpecAsString(object[[1]]))
+})
+
+#' StuydIdColName as String
+#' 
+#' This function returns the studyIdColName from an StudySpecificVocabulary
+#' 
+#' @param object veupathUtils::StudySpecificVocabulary
+#' @return character
+#' @export
+setGeneric("getStudyIdColName", 
+  function(object) standardGeneric("getStudyIdColName"),
+  signature = "object"
+)
+
+#' @export
+setMethod('getStudyIdColName', signature('StudySpecificVocabulary'), function(object) {
+  return(object@studyIdColName)
+})
+
+#' @export
+setMethod('getStudyIdColName', signature('StudySpecificVocabulariesByVariable'), function(object) {
+  #since we validate theyre all the same, can just take the first
+  return(object[[1]]@studyIdColName)
+})
+
+#' as.data.table
+#' 
+#' This function returns a data.table representation of 
+#' StudySpecificVocabulary or StudySpecificVocabularyByVariable
+#' 
+#' @return data.table
+#' @export 
+as.data.table <- makeGeneric('as.data.table', data.table::as.data.table)
+
+#' @export
+setMethod('as.data.table', signature('StudySpecificVocabulary'), function(x) {
+  .dt <- data.table::data.table('study'=x@study, 'variable'=x@vocabulary)
```

Comment:
so we're just making a tiny data table, yes? Since `study` and `vocabulary are both chaaracters?

Reply:
oh i see the next line. Makes much more sense now :)